xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-oldcore/src/main/aspect/com/xpn/xwiki/web/DeleteActionCompatibilityAspect.aj

```diff
@@ -58,6 +58,6 @@ public privileged aspect DeleteActionCompatibilityAspect
     public boolean DeleteAction.deleteToRecycleBin(EntityReference entityReference, XWikiContext context)
         throws XWikiException
     {
-        return this.deleteDocument(entityReference, context, false);
+        return this.deleteDocument(entityReference, context, false, null, false, false);
```

Comment:
This is sending a warning bell to me as we're not supposed to touch legacy code, and it usually means that some public API was broken.